flag-app/flag-app-project/src/hooks/useApiData.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react";
+
+import apiClient from "../services/api-client";
+import { CanceledError } from "axios";
+import { Country } from "../interfaces";
+
+export const useApiData = (searchedQuery?: string) => {
+  const [countries, setCountries] = useState<Country[]>([]);
+  const [error, setError] = useState("");
+  const [isLoading, setLoading] = useState(false);
+
+  useEffect(() => {
```

Comment:
This is bad practise. I strongly recommend you check this out and implement this using React Query:

https://www.youtube.com/watch?v=vxkbf5QMA2g